aggsender/db/aggsender_db_storage.go

```diff
@@ -597,11 +577,77 @@ func (a *AggSenderSQLStorage) GetNonAcceptedCertificate() (*NonAcceptedCertifica
 	return &nonAcceptedCert, nil
 }
 
+// Move to certificate_info_history the certificate identified by CertificateKey
+func (a *AggSenderSQLStorage) MoveCertificateToHistory(tx dbtypes.Querier, height uint64) error {
+	a.logger.Debugf("moving certificate to history - height: %d", height)
+	if _, err := tx.Exec(`INSERT INTO certificate_info_history SELECT * FROM certificate_info WHERE height = $1;`,
+		height); err != nil {
+		return fmt.Errorf("error moving certificate height: %d to history: %w", height, err)
+	}
+	return a.deleteCertificates(tx, tableCertificate, height)
+}
+
+// Delete from certificate_info and certificate_info_history the certificate CertificateKey
+// if you don't need a tx just pass nil
+// It required to be in certificate_info table, if not found it returns ErrNoCertDeleted error
+func (a *AggSenderSQLStorage) DeleteCertificate(tx dbtypes.Querier, height uint64, mustDelete DeleteFlag) error {
+	if tx == nil {
+		tx = a.db
+	}
+	// If there are no certificates in cert table, we return an error
+	if err := a.deleteCertificates(tx, tableCertificate, height); err != nil {
+		// If we allow no rows affected we ignore the error ErrNoCertDeleted
+		if !errors.Is(err, ErrNoCertDeleted) || mustDelete {
+			return fmt.Errorf("error deleting certificate height %d from cert table: %w", height, err)
+		}
+	}
+	// If there are no certificates in history table, we ignore the error
+	if err := a.deleteCertificates(tx, tableCertificateHistory, height); err != nil && !errors.Is(err, ErrNoCertDeleted) {
+		return fmt.Errorf("error deleting certificate height %d from cert history table: %w", height, err)
+	}
+	return nil
+}
+
+// deleteCertificateHistory deletes the certificate history for the given height
+func (a *AggSenderSQLStorage) deleteCertificates(tx dbtypes.Querier, table tableName, height uint64) error {
+	certInfos, err := getCertificatesByHeight(tx, table, height)
+	if err != nil {
+		return fmt.Errorf("error deleting certificate history height %d when reading it: %w", height, err)
+	}
+	if len(certInfos) == 0 {
+		return fmt.Errorf("deleteCertificates no certificates found for height %d in table %s: %w", height, table,
+			ErrNoCertDeleted)
+	}
+	for _, certInfo := range certInfos {
+		a.deleteCertificateFile(certInfo.SignedCertificateFilename())
+	}
+	if _, err := tx.Exec(fmt.Sprintf(`DELETE FROM %s WHERE height = $1;`, table), height); err != nil {
+		return fmt.Errorf("error deleting certificate history height %d: %w", height, err)
+	}
+	return nil
+}
+
+// Delete from certificate_info and certificate_info_history all certificates older than olderThanHeight
+func (a *AggSenderSQLStorage) DeleteOldCertificates(tx dbtypes.Querier, olderThanHeight uint64) error {
```

Comment:
WDYT, slightly shorter but still understandable (or even `cutoffHeight`)?
```suggestion
func (a *AggSenderSQLStorage) DeleteOldCertificates(tx dbtypes.Querier, maxHeight uint64) error {
```